lib/web_ui/dev/driver_manager.dart

```diff
@@ -20,7 +20,34 @@ import 'utils.dart';
 ///
 /// This manager can be used for both macOS and Linux.
 class ChromeDriverManager extends DriverManager {
-  ChromeDriverManager(String browser) : super(browser);
+
+  /// Directory which contains the preferred version of Chrome.
+  ///
+  /// On LUCI we are using the CIPD packages to control Chrome binaries we use.
+  /// There will be multiple CIPD packages loaded at the same time. Yaml file
+  /// `driver_version.yaml` contains the version number we want to use.
+  ///
+  /// Local integration tests are still using the system Chrome. This should be
+  /// fixed in issue: https://github.com/flutter/flutter/issues/53179.
+  ///
+  /// Initialized to the current first to avoid the `Non-nullable` error.
```

Comment:
If `io.Directory.current` is not a valid value for this field, consider making it `late final` and leaving it uninitialized. We don't want to accidentally write into the current working directory. A null error is better.

Reply:
thanks! I didn't thought of `late`. It's very useful

Reply:
done.

Reply:
Reverting this, unfortunately it cause errors. My guess is it is not meant to be used for child classes. May be if this was a property on the parent class it might have worked. However, it is not relevant to other browsers like Safari.

```
dev/driver_manager.dart:31:3: Error: Variables must be declared using the keywords 'const', 'final', 'var' or a type name.
Try adding the name of the type of the variable or the keyword 'var'.
  late final io.Directory _browserDriverDirWithVersion;
  ^^^^
dev/driver_manager.dart:31:3: Error: Expected ';' after this.
  late final io.Directory _browserDriverDirWithVersion;
  ^^^^
dev/driver_manager.dart:41:5: Error: The setter '_browserDriverDirWithVersion' isn't defined for the class 'ChromeDriverManager'.
 - 'ChromeDriverManager' is from 'dev/driver_manager.dart'.
Try correcting the name to the name of an existing setter, or defining a setter or field named '_browserDriverDirWithVersion'.
    _browserDriverDirWithVersion = io.Directory(pathlib.join(
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
dev/driver_manager.dart:31:27: Error: Final field '_browserDriverDirWithVersion' is not initialized.
Try to initialize the field in the declaration or in every constructor.
  late final io.Directory _browserDriverDirWithVersion;
```

Reply:
interesting even only using `late io.Directory _browserDriverDirWithVersion;` gives an error. Even though it's a common usage in other parts of the engine.
```
dev/driver_manager.dart:32:3: Error: Type 'late' not found.
  late io.Directory _browserDriverDirWithVersion;
```

Reply:
Ah, this file hasn't been migrated to non-nullable Dart yet. In that case let's back up a bit. What did you mean "avoid the `Non-nullable` error"? What did this error look like?

Reply:
For the current code, VS code gives the following error (underlining the constructor with red):
```
Non-nullable instance field '_browserDriverDirWithVersion' must be initialized.
Try adding an initializer expression, or add a field initializer in this constructor, or mark it 'late'.
```
It does not break the build since we didn't migrate these test harness files yet.

Reply:
Looks like `web_engine_analysis` also breaks:

```
+ echo 'Engine path /b/s/w/ir/cache/builder'
+ DART_SDK_DIR=/b/s/w/ir/cache/builder/src/out/host_debug_unopt/dart-sdk
+ PUB_PATH=/b/s/w/ir/cache/builder/src/out/host_debug_unopt/dart-sdk/bin/pub
+ DART_ANALYZER_PATH=/b/s/w/ir/cache/builder/src/out/host_debug_unopt/dart-sdk/bin/dartanalyzer
+ echo 'Running `pub get` in '\''engine/src/flutter/lib/web_ui'\'''
+ cd ''
+ /b/s/w/ir/cache/builder/src/out/host_debug_unopt/dart-sdk/bin/pub get
Engine path /b/s/w/ir/cache/builder
Running `pub get` in 'engine/src/flutter/lib/web_ui'
Resolving dependencies...
Got dependencies!
Running `dartanalyzer` in 'engine/src/flutter/lib/web_ui'
+ echo 'Running `dartanalyzer` in '\''engine/src/flutter/lib/web_ui'\'''
+ cd ''
+ /b/s/w/ir/cache/builder/src/out/host_debug_unopt/dart-sdk/bin/dartanalyzer --enable-experiment=non-nullable --fatal-warnings --fatal-hints dev/ lib/ test/ tool/
Analyzing dev, lib, test, tool...
  error • Non-nullable instance field '_browserDriverDirWithVersion' must be initialized. • dev/driver_manager.dart:34:3 • not_initialized_non_nullable_instance_field
1 error found.
```
[link](https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8871603660520423824/+/steps/web_engine_analysis/0/stdout)

Reply:
Ah, I know what's happening. This file needs `// @dart = 2.6` just before the first import ([example](https://github.com/flutter/engine/blob/master/lib/web_ui/dev/create_simulator.dart#L5)). Without that line this file is treated as null-safe, even though it imports libraries that haven't migrated to null safety yet.

Reply:
Thanks, I'll add it. I never noticed it was missing. Let me check the other files to see if there are any other missing ones.

Reply:
done. I also updated one more file.